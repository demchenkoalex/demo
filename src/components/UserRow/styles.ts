import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    marginHorizontal: 16,
    marginTop: 32
  },
  firstNameText: {
    fontSize: 16,
    fontWeight: '600'
  },
  image: {
    borderRadius: 20,
    height: 40,
    width: 40
  },
  lastNameText: {
    fontSize: 14
  },
  textContainer: {
    flex: 1,
    justifyContent: 'center',
    marginLeft: 16
  }
})
