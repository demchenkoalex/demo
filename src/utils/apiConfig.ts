import axios from 'axios'

const api = axios.create({
  baseURL: 'https://reqres.in/api',
  timeout: 10000
})

export default api
