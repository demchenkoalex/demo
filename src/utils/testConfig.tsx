import React, { ComponentType } from 'react'
import { createRenderer } from 'react-test-renderer/shallow'

export const setupJestShallow = ({ Component, props }: { Component: ComponentType<any>, props: any }) => {
  const shallowRenderer = createRenderer()

  shallowRenderer.render(<Component {...props} />)

  const wrapper = shallowRenderer.getRenderOutput()
  const instance = shallowRenderer.getMountedInstance() as any

  return {
    wrapper: { ...wrapper, instance: () => instance }
  }
}
