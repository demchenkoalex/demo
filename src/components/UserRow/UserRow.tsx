import React from 'react'
import { Image, ImageStyle, Text, View } from 'react-native'
import { User } from 'src/types'
import styles from './styles'

export interface Props {
  user: User
}

export const UserRow = ({ user }: Props) => (
  <View style={styles.container}>
    <Image style={styles.image as ImageStyle} source={{ uri: user.avatar }} />
    <View style={styles.textContainer}>
      <Text style={styles.firstNameText} numberOfLines={1}>{user.first_name}</Text>
      <Text style={styles.lastNameText} numberOfLines={1}>{user.last_name}</Text>
    </View>
  </View>
)
