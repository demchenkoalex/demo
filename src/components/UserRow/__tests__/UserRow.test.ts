import { setupJestShallow } from 'src/utils/testConfig'
import { Props, UserRow } from '../UserRow'

describe('UserRow', () => {
  const props: Props = {
    user: {
      avatar: 'https://avatar.com/alex',
      first_name: 'Alex',
      id: 1,
      last_name: 'Demchenko'
    }
  }

  describe('Shapshots', () => {
    it('with default props', () => {
      const { wrapper } = setupJestShallow({
        Component: UserRow,
        props
      })

      expect(wrapper).toMatchSnapshot()
    })
  })
})
