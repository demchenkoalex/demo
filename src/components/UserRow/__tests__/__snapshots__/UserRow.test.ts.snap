// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRow Shapshots with default props 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginHorizontal": 16,
      "marginTop": 32,
    }
  }
>
  <Component
    source={
      Object {
        "uri": "https://avatar.com/alex",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 16,
      }
    }
  >
    <Component
      numberOfLines={1}
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Alex
    </Component>
    <Component
      numberOfLines={1}
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      Demchenko
    </Component>
  </Component>
</Component>
`;
